test/EventSourcing/SnapshottingEventSourcingRepositoryTest.php

```diff
@@ -0,0 +1,152 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Broadway\Snapshotting\EventSourcing;
+
+use Broadway\Domain\DomainEventStream;
+use Broadway\Domain\DomainMessage;
+use Broadway\Domain\Metadata;
+use Broadway\EventSourcing\EventSourcedAggregateRoot;
+use Broadway\EventSourcing\EventSourcingRepository;
+use Broadway\EventStore\EventStore;
+use Broadway\Snapshotting\EventSourcing\Testing\TestEventSourcedAggregateRoot;
+use Broadway\Snapshotting\Snapshot\Snapshot;
+use Broadway\Snapshotting\Snapshot\SnapshotRepository;
+use Broadway\Snapshotting\Snapshot\Trigger\EventCountTrigger;
+use PHPUnit_Framework_TestCase;
+use Prophecy\Argument;
+
+class SnapshottingEventSourcingRepositoryTest extends PHPUnit_Framework_TestCase
+{
+    private $eventStore;
+    private $eventSourcingRepository;
+    private $snapshotRepository;
+    private $snapshottingEventSourcingRepository;
+
+    public function setUp()
+    {
+        $this->eventSourcingRepository = $this->prophesize(EventSourcingRepository::class);
+        $this->eventStore              = $this->prophesize(EventStore::class);
+        $this->snapshotRepository      = $this->prophesize(SnapshotRepository::class);
+
+        $this->snapshottingEventSourcingRepository = new SnapshottingEventSourcingRepository(
+            $this->eventSourcingRepository->reveal(),
+            $this->eventStore->reveal(),
+            $this->snapshotRepository->reveal(),
+            new EventCountTrigger(100)
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_reconstitutes_aggregate_when_no_snapshot_found()
+    {
+        $this->snapshotRepository
+            ->load(42)
+            ->shouldBeCalled()
+            ->willReturn(null)
+        ;
+
+        $this->eventSourcingRepository
+            ->load(42)
+            ->shouldBeCalled()
+            ->willReturn($this->prophesize(EventSourcedAggregateRoot::class)->reveal())
+        ;
+
+        $this->snapshottingEventSourcingRepository->load(42);
+    }
+
+    /**
+     * @test
+     */
+    public function it_queries_the_event_store_for_events_recorded_after_playhead_of_snapshot()
+    {
+        // create a snapshot of an aggregate root with 5 committed events
+        $snapshot = $this->createSnapshot();
+        $this->assertEquals(4, $snapshot->getPlayhead());
+
+        $this->snapshotRepository
+            ->load(42)
+            ->shouldBeCalled()
+            ->willReturn($snapshot)
+        ;
+
+        $this->eventSourcingRepository
+            ->load(42)
+            ->shouldNotBeCalled()
+        ;
+
+        // 2 new events since last snapshot
+        $domainEventStream = new DomainEventStream([
+            DomainMessage::recordNow(42, 5, new Metadata([]), new \stdClass()),
+            DomainMessage::recordNow(42, 6, new Metadata([]), new \stdClass()),
+        ]);
+
+        $this->eventStore
+            ->loadFromPlayhead(42, 4)
```

Comment:
Should be changed to `42, 5` to check the snapshot playhead + 1 loading. Based on the `EventStoreTest` implementation in broadway, a GTEQ comparison is expected.